formats/notion/convert-to-md.ts

```diff
@@ -0,0 +1,257 @@
+import { htmlToMarkdown } from 'obsidian';
+import {
+	escapeHashtags,
+	escapeRegex,
+	getParentFolder,
+	matchFilename,
+} from '../../util';
+import {
+	assembleParentIds,
+	getNotionId,
+	stripNotionId,
+	parseDate,
+} from './notion-utils';
+import moment from 'moment';
+
+export function convertNotesToMd({
+	idsToFileInfo,
+	pathsToAttachmentInfo,
+	attachmentFolderPath,
+}: {
+	idsToFileInfo: Record<string, NotionFileInfo>;
+	pathsToAttachmentInfo: Record<string, NotionAttachmentInfo>;
+	attachmentFolderPath: string;
+}) {
+	for (let fileInfo of Object.values(idsToFileInfo)) {
+		convertLinksToObsidian(fileInfo.notionLinks, {
+			idsToFileInfo,
+			pathsToAttachmentInfo,
+		});
+
+		replaceNestedTags(fileInfo.body, 'strong');
+		replaceNestedTags(fileInfo.body, 'em');
+		stripLinkFormatting(fileInfo.body);
+		encodeNewlinesToBr(fileInfo.body);
+		fixNotionDates(fileInfo.body);
+		fixNotionLists(fileInfo.body);
+		replaceTableOfContents(fileInfo.body);
+
+		let htmlString = fileInfo.body.innerHTML;
+		// Simpler to just use the HTML string for this replacement
+		splitBrsInFormatting(htmlString, 'strong');
+		splitBrsInFormatting(htmlString, 'em');
+
+		fileInfo.markdownBody = htmlToMarkdown(htmlString);
+		fileInfo.markdownBody = escapeHashtags(fileInfo.markdownBody);
+
+		if (fileInfo.properties) {
+			fileInfo.yamlProperties = fileInfo.properties.map((property) =>
+				convertPropertyToYAML(property, {
+					fileInfo,
+					idsToFileInfo,
+					pathsToAttachmentInfo,
+					attachmentFolderPath,
+				})
+			);
+		}
+	}
+}
+
+const replaceNestedTags = (body: HTMLElement, tag: 'strong' | 'em') => {
+	body.querySelectorAll(tag).forEach((el) => {
+		if (el.parentElement.tagName === tag.toUpperCase()) return;
+		const nestedEls = el.querySelectorAll(tag);
+		if (nestedEls.length === 0) return;
+		for (let i = 0; i < nestedEls.length; i++) {
```

Comment:
Hmm I'm not sure if this does what we think it's trying to do - it might be doing something unexpected, for example, if the nesting goes more than one level deep (i.e. it skips levels down). Seems like this will hoist the nested children all the way up to the root level tag. For example if we had
```html
<b>
<i><b>test</b></i>
</b>
```
this code will move `test` directly into the top level `<b>` .

Perhaps what should happen here is that the childNodes get moved to a DocumentFragment, and we use `nestedEls[i].replaceWith(fragment)`.

Reply:
that sounds good!